odoo/models.py

```diff
@@ -241,12 +246,11 @@ def __new__(meta, name, bases, attrs):
                     f"Invalid import of {module}.{name}, it should start with 'odoo.addons'."
                 attrs['_module'] = module.split('.')[2]
 
-            # determine model '_name' and normalize '_inherits'
-            inherit = attrs.get('_inherit', ())
-            if isinstance(inherit, str):
-                inherit = attrs['_inherit'] = [inherit]
-            if '_name' not in attrs:
-                attrs['_name'] = inherit[0] if len(inherit) == 1 else name
+            if not attrs.get('_name'):
+                attrs['_name'] = class_name_to_model_name(name)
+
+            if isinstance(attrs.get('_inherit'), str):
+                raise TypeError(f"'_inherit' property of model {name!r} should be a list: {attrs['_inherit']!r}.")
```

Comment:
The message doesn't match the assertion

```suggestion
            _inherit = attrs.get('_inherit')
            if _inherit and not isinstance(_inherit, list):
                raise TypeError(f"'_inherit' property of model {name!r} should be a list: {_inherit!r}.")
```